lib/custom_set_cases.rb

```diff
@@ -0,0 +1,89 @@
+class CustomSetCase < OpenStruct
+  def name
+    'test_%s' % description.gsub(/ |-/, '_')
+  end
+
+  def test_body
+    instance_eval section
```

Comment:
An instance_eval, but yes.  Section is the hash being generated from the JSON.  Should use parse here.  It was definitely a 'get it done' and 'dirty'.